app/test/main_test.dart

```diff
@@ -1,10 +1,18 @@
+import 'package:app/core/services/service_locator.dart';
 import 'package:app/humane_transport_app.dart';
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
-  testWidgets('HumaneTransportApp says Welcome', (WidgetTester tester) async {
-    await tester.pumpWidget(HumaneTransportApp());
-    final textFinder = find.text('Welcome');
-    expect(textFinder, findsOneWidget);
+  group('Group test', () {
+    // Needed to show where the implementations are located
+    setUpAll(() {
+      setUpLocator();
```

Comment:
We'll likely have a test version of this which implements the locator. We'll shove mocks into it.

- [ ] Might as well make our setUpLocator() mockable -> make it so main gives it things like AuthenticationService and such (give it a constructor)

We will end up having two mains: our main-main which is dumb and gives static things to GetIt, and our testing main which takes in dependencies and sets up it's getIt with them. This way the we can make all app dependencies mockable at once!!

Reply:
I plan to this next time. I already have task for API in my queue. I will do this after API task

Reply:
You should add tests as part of API or #130 . We absolutely require tests on non-experimental code.